scripts/cfengine-build-host-setup.cf

```diff
@@ -0,0 +1,188 @@
+      "build-essential";
+      "curl"; # added for debian-10
+      "debhelper";
+      "dpkg-dev";
+      "expat";
+      "fakeroot";
+      "flex";
+      "gdb";
+      "libncurses5"; # added for debian-10
+      "libncurses5-dev"; # added for debian-10
+      "libexpat1-dev";
+      "libmodule-load-conditional-perl";
+      "libpam0g-dev";
+      "ntp";
+      "pkg-config";
+      "psmisc";
+      "python3-pip";
+      "rsync"; # added for debian-10
+
+# I attempted to arrange these packages in order of: generic (all versions) and then as if we gradually added them through time: rhel-6, 7, 8, 9...
+    suse|opensuse|sles|redhat|centos::
+      "gcc";
+      "ncurses-devel";
+      "pam-devel";
+      "rsync";
+      "make";
+      "rpm-build";
+
+    redhat|centos::
+      "expat-devel";
+      "gcc-c++";
+      "gettext";
+      "ncurses";
+      "perl-Digest-MD5";
+      "perl-Module-Load-Conditional";
+      "wget";
+
+    redhat_6|centos_6|redhat_7|centos_7::
+      "gdb";
+      "java-11-openjdk"; # ok for jenkins 2.426.1 https://www.jenkins.io/doc/book/platform-information/support-policy-java/index.html
+      "ntp";
+      "pkgconfig";
+      "perl-IO-Uncompress-Gunzip";
+      "perl-JSON-PP";
+      "perl-Data-Dumper";
+      "perl-IPC-Cmd";
+      "perl-devel";
+      "python-psycopg2";
+      "xfsprogs";
+
+    redhat_7|centos_7::
+      "ccache";
+      "fakeroot";
+      "shellcheck"; # for what purpose? and why not redhat_6? unused platform? also not rhel 8 and 9 so ???
+
+    redhat_7|centos_7|redhat_9::
+      "python3-pip";
+
+    redhat_7|centos_7|redhat_8|centos_8|redhat_9::
+      "perl-ExtUtils-MakeMaker";
+      "psmisc";
+      "python3"; # what does redhat_6 get?
+      "tmux"; # why?! :O
+      "which";
+
+    redhat_8|centos_8::
+      "python2";
+      "python2-psycopg2";
+
+    redhat_8|centos_8|redhat_9::
+      "perl";
+      "pkgconf"; # pkgconfig renamed to pkgconf in rhel8
+      "selinux-policy-devel"; # maybe add to _7 and _6?
+      "openssl-devel";
+
+    redhat_9::
+      "python3-psycopg2";
+
+    suse|opensuse|sles::
+      "binutils";
+      "pam";
+      "pkg-config";
+      "patch";
+      "gdb";
+      "java-17-openjdk";
+
+
+  vars:
+    "suse_users_and_groups" slist => { "daemon", "bin", "sys" };
+  classes:
+    debian_9|ubuntu_16::
+      "have_opt_jdk21" expression => fileexists("/opt/jdk-21.0.1");
+    redhat_7|centos_7|redhat_8|centos_8|redhat_9::
+      "have_development_tools" expression => returnszero("yum groups list --installed | grep 'Development Tools' >/dev/null", "useshell");
+    redhat_8|centos_8::
+      "have_fakeroot" expression => returnszero("command -v fakeroot >/dev/null", "useshell");
+
+  commands:
+    !have_opt_jdk21.(debian_9|ubuntu_16)::
+      "sh $(this.promise_dirname)/linux-install-jdk21.sh" contain => in_shell;
+    (redhat_7|centos_7|redhat_8|centos_8|redhat_9).!have_development_tools::
+      "yum groups install 'Development Tools'" contain => in_shell;
```

Comment:
package_method => yum_group might work but this works "for sure" :)

Reply:
yep, package_method => yum_group works but is noisy... always tries to reinstall. reverted. back to classes/command strategy.